src/qibo/gates/gates.py

```diff
@@ -2,11 +2,89 @@
 from typing import List
 
 import numpy as np
+import sympy as sp
 
 from qibo.config import PRECISION_TOL, raise_error
 from qibo.gates.abstract import Gate, ParametrizedGate
 
 
+class Parameter:
+    """Object which allows complex gate parameters. Several trainable parameter
```

Comment:
I guess that here by "complex" you mean that there is some classical pre-processing in the parameter calculation. The word may be a bit confusing as one can think of complex (with real and imaginary part) numbers, but this may just be me.